### PR TITLE
show delete confirmation dialog

### PR DESCRIPTION
**Pull request by mslourens**
implements #1320

### CODE REVIEW

**Comment by sferra** [comment]
@mslourens What do you think about shortening the message to simply "Move the note to the trash?" and changing the "Confirm" button text to "Move" for the sake of simplictity?

Also do you think that there is maybe some way to de-duplicate the code without any major refactoring?

**Comment by kazup01** [comment]
Hi @mslourens , thanks for your support . Could you fix CI error? https://travis-ci.org/BoostIO/Boostnote/jobs/319797454

File: browser/main/Detail/MarkdownNoteDetail.js [hunk 1]
```diff
@@ -176,36 +176,34 @@ class MarkdownNoteDetail extends React.Component {
   handleTrashButtonClick (e) {
     const { note } = this.state
     const { isTrashed } = note
+    const { confirmDeletion } = this.props
 
     if (isTrashed) {
-      const dialogueButtonIndex = dialog.showMessageBox(remote.getCurrentWindow(), {
```
Comment (sosukesuzuki) [inline browser/main/Detail/MarkdownNoteDetail.js]: dialog is never used.
I think import is unnecessary on line 31.

File: browser/main/Detail/MarkdownNoteDetail.js [hunk 2]
```diff
@@ -176,36 +176,34 @@ class MarkdownNoteDetail extends React.Component {
...
-        this.save()
-      })
+        this.setState({
+          note
+        }, () => {
+          this.save()
+        })
+      }
     }
     ee.emit('list:next')
```
Comment (sosukesuzuki) [inline browser/main/Detail/MarkdownNoteDetail.js]: I want you to make don't call `ee.emit('line:next')` when I push `cancel`. 
(please fix it too on `SnippetNoteDetail.js`.)

**Comment by mslourens** [comment]
sorry for the delay, I haven't been working for some time, because of the holidays.

File: browser/main/Detail/MarkdownNoteDetail.js [hunk 3]
```diff
@@ -28,7 +28,6 @@ import striptags from 'striptags'
 
 const electron = require('electron')
 const { remote } = electron
```
Comment (sosukesuzuki) [inline browser/main/Detail/MarkdownNoteDetail.js]: `remote` and `electron` are unnecessary.

File: browser/main/Detail/MarkdownNoteDetail.js [hunk 4]
```diff
@@ -203,9 +202,11 @@ class MarkdownNoteDetail extends React.Component {
         }, () => {
           this.save()
         })
+
+        ee.emit('list:next')
       }
     }
-    ee.emit('list:next')
+
```
Comment (sosukesuzuki) [inline browser/main/Detail/MarkdownNoteDetail.js]: please remove this blank line.

**Comment by sosukesuzuki** [comment]
Sorry for delay.
I approved! LGTM!

**Comment by kazup01** [comment]
Merged. Thanks @mslourens !

**Pull request merged**